### PR TITLE
Select text on SpinBox on double click

### LINKED ISSUE #6104: Double-clicking text in `SpinBox` should select it
In `LineEdit`, double-clicking selects a word. `SpinBox` should behave the same. However, of the text `-10` in a `LineEdit`, just `10` is selected when double-clicking it. A `SpinBox` should probably select the whole negative number.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #6104

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ All notable changes to this project are documented in this file.
  - Do not trigger `current-item-changed` on `StandardListView` if `current-item` is set on the same value.
  - Fixed `TimePickerPopup` does not open minute view by click on selected hour.
  - Visually clamp the slider even if the value is out of bounds (#5770)
+ - Fixed select text in `SpinBox` on double click (#6104)
```
Comment (tronical) [inline CHANGELOG.md:30]:
```suggestion
 - Fixed text in `SpinBox` not being selected on double click (#6104)
```

File: internal/compiler/widgets/common/spinbox-base.slint [hunk 2]
```diff
@@ -87,6 +87,11 @@ export component SpinBoxBase {
             return reject;
         }
 
+        double-clicked => {
```
Comment (tronical) [inline internal/compiler/widgets/common/spinbox-base.slint:90]: This fixes that particular issue, but it makes me wonder: There are more click behaviours on the underlying `TextInput` that should be exposed, but aren't - such as triple click. And context menu in the future.

I think that perhaps this `TouchArea` should be placed underneath the `TextInput` and only deal with the scroll event. The existing `clicked` and the new `double-clicked` should already be handled by `TextInput` natively, no?

**Review by tronical — APPROVED** [review]
It seems the basic functionality of click focusing the spinbox is covered by the existing tests/cases/widgets/spinbox_basic.slint 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -28,6 +28,8 @@ All notable changes to this project are documented in this file.
  - Fixed `TimePickerPopup` does not open minute view by click on selected hour.
  - Visually clamp the slider even if the value is out of bounds (#5770)
  - Fixed button of Cupertino `ComboBox` not centered when height is bigger than default.
+ - Fixed select text in `SpinBox` on double click (#6104)
+ - Fixed text in `SpinBox` not being selected on double click (#6104)
```
Comment (ogoffart) [inline CHANGELOG.md:32]: Why is this line twice?

**Review by ogoffart — COMMENTED** [review]
(There could have been a test)

**Pull request merged**